integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py

```diff
@@ -267,6 +267,19 @@ def _embedding_retrieval(
 
         return self._convert_query_result_to_documents(result)
 
+    @staticmethod
+    def _convert_to_int(metadata: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Convert metadata original int values back to integers again, since Pinecone store metadata numbers as float.
```

Comment:
Let's explain we only do this for specific metadata fields used by some retrievers.

Reply:
I need your help in understanding what's missing, it's explicitly on the docstring why and how is used, even being a private function.

Please add here, in the comment, the exact wording you would like to see, and I change it.